src/Multisite_Control.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * @license GPL-3.0-or-later
+ *
+ * Modified by Justin Vogt on 27-August-2024 using {@see https://github.com/BrianHenryIE/strauss}.
+ */
+
+namespace juvo\Bricks_Custom_Queries;
+
+class Multisite_Control
+{
+
+    private string $name;
+    private string $label;
+
+    public function __construct(string $name, string $label)
+    {
+        $this->name = $name;
+        $this->label = $label;
+    }
+
+    /**
+     * @param array $args
+     * @param $query_obj
+     * @param Query_Type $type
+     * @return array
+     */
+    public function add_multisite_arg(array $args, $query_obj, Query_Type $type): array {
+
+        if (empty($query_obj->settings['blog_id_'.$this->name])) {
+            return $args;
+        }
+
+        switch ($type) {
+            case Query_Type::Post:
+                $args['blog_id'] = $query_obj->settings['blog_id_'.$this->name];
+                break;
+            default:
+                break;
+        }
```

Comment:
Not not for the other types too? What about users and terms? How do we handle these?

Reply:
Actually, I removed the Query-Type Stuff because there is no need. If it's a multisite switch, then its for everything. We don't need to care about the types. Therefore changed.